### PR TITLE
Add support to Dart http client

### PR DESCRIPTION
**Pull request by Husseinhj**
image
-----
### Feature:
This is the [Dart language](https://dart.dev/) plugin that has support from the [http](https://pub.dev/packages/http) client.

### Related issues:
Dart code generation #231
dart support in code generator #157

### CODE REVIEW

**Comment by Husseinhj** [comment]
Hi @abhijitkane 
Do you have any update to review this pull request? 😊

**Comment by Husseinhj** [comment]
@abhijitkane @umeshp7

**Comment by Husseinhj** [comment]
Kindly reminder @abhijitkane @umeshp7

**Review by akacokafor — APPROVED** [review]

**Comment by Husseinhj** [comment]
Thanks @akacokafor 👍

**Comment by Husseinhj** [comment]
Kindly reminder @abhijitkane @umeshp7 @shreys7 @sastava007

**Comment by Husseinhj** [comment]
Kindly reminder @abhijitkane @umeshp7 @shreys7 @sastava007 @webholik

File: codegens/dart/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (umeshp7) [inline codegens/dart/package.json:1]: The name of the folder for this dart-http. 
also for the name here in the package json should include its variant in the name.

File: codegens/dart/lib/index.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+          contentType = param.contentType;
+        if (type === 'file') {
+          if (typeof param.src !== 'string') {
+            if (Array.isArray(param.src) && param.src.length) {
+              param.src.forEach((filePath) => {
+                addFormParam(formdataArray, key, param.type, filePath, disabled, contentType);
+              });
+            }
+            else {
+              addFormParam(formdataArray, key, param.type, '/path/to/file', disabled, contentType);
```
Comment (umeshp7) [inline codegens/dart/lib/index.js:221]: https://github.com/postmanlabs/postman-code-generators/blob/develop/codegens/nodejs-native/lib/request.js#L82 

All these branches need proper comments as mentioned here^

**Review by umeshp7 — CHANGES_REQUESTED** [review]

**Comment by umeshp7** [comment]
The new folder for the Dart code gen should be called Dart-http.

File: codegens/dart/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        formDataFileArray.push(`request.files.add(await http.MultipartFile.fromPath('${key}', '${data.src}'));`);
+      }
+      else {
+        formDataArray.push(`${indent}'${sanitize(key)}': '${sanitize(value, trim)}'`);
+      }
+    }
+  });
+
+  if (formDataArray.length > 0) {
+    bodySnippet += 'request.fields.addAll({\n';
```
[the remark this answers is not in the document]
Comment (webholik) [reply codegens/dart/lib/index.js:95]: @umeshp7 We have a line in the convert fucntion which handles indentation:
```
    //  if boilerplate is included then two more indent needs to be added in snippet
    (options.includeBoilerplate) &&
    (codeSnippet = indent + codeSnippet.split('\n').join('\n' + indent) + '\n');
```
Indentation is only required if the snippet is inside the main function.

File: codegens/dart/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+  if (_.isEmpty(body)) {
+    return bodySnippet;
+  }
+
+  _.forEach(body, function (data) {
+    key = trim ? data.key.trim() : data.key;
+    value = trim ? data.value.trim() : data.value;
+    if (!data.disabled) {
+      if (data.type === 'file') {
+        formDataFileArray.push(`request.files.add(await http.MultipartFile.fromPath('${key}', '${data.src}'));`);
```
[the remark this answers is not in the document]
Comment (webholik) [reply codegens/dart/lib/index.js:86]: Same as above

File: codegens/dart/lib/index.js (hunk 2 above)
Comment (webholik) [reply codegens/dart/lib/index.js]: Done

File: codegens/dart/package.json (hunk 1 above)
Comment (webholik) [reply codegens/dart/package.json]: Done

File: codegens/dart/test/unit/convert.test.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+var convert = require('../../index').convert,
```
[the remark this answers is not in the document]
Comment (webholik) [reply codegens/dart/test/unit/convert.test.js]: Done

**Review by umeshp7 — APPROVED** [review]

**Pull request merged**